### PR TITLE
Add armstrong-numbers exercise

### PR DESCRIPTION
**Pull request by PakkuDon**
[Problem specification and test data](https://github.com/exercism/problem-specifications/tree/master/exercises/armstrong-numbers)

### CODE REVIEW

**Comment by ajomadlabs** [comment]
@PakkuDon Correct the linting errors

File: exercises/armstrong-numbers/example.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = {
+  validate: function(input) {
```
Comment (ajomadlabs) [inline exercises/armstrong-numbers/example.js]: Space before function parenthesis `funcntion (input)`

File: exercises/armstrong-numbers/example.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    var digits = (input + '').split('');
+    var sum = digits.reduce(function (total, current) {
+      return total += Math.pow(current, digits.length);
```
Comment (ajomadlabs) [inline exercises/armstrong-numbers/example.js]: do not the assign during return
`total += Math.pow(current, digits.length);`
 `return total`

File: exercises/armstrong-numbers/example.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    }, 0);
+    return sum == input;
+  },
```
Comment (ajomadlabs) [inline exercises/armstrong-numbers/example.js]: Remove the comma

**Comment by PakkuDon** [comment]
@ajomadlabs Lint fixes applied in ce16af8. Sorry about that

File: exercises/armstrong-numbers/example.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = {
+  validate: function (input) {
+    var digits = (input + '').split('');
+    var sum = digits.reduce(function (total, current) {
+      return total + Math.pow(current, digits.length);
```
Comment (ajomadlabs) [inline exercises/armstrong-numbers/example.js]: Do assign while returning
`total + Math.pow(current, digits.length);`
`return total`

File: exercises/armstrong-numbers/example.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = {
+  validate: function (input) {
+    var digits = (input + '').split('');
```
Comment (matthewmorgan) [inline exercises/armstrong-numbers/example.js]: Similar to my comment on the ES submission, let's use `String(input)` here to make it clear what is happening.

Comment (PakkuDon) [reply exercises/armstrong-numbers/example.js]: Done, amended into the previous commit 👍

**Comment by matthewmorgan** [comment]
@PakkuDon thanks for your work!

**Pull request merged**